portfolio/src/main/webapp/script.js

```diff
@@ -25,3 +25,30 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 */
+
+
+/**
+ * Fetches greeting from the server and adds it to the DOM.
+ */
+function getGreeting() {
+  console.log('Fetching a greeting.');
+  const responsePromise = fetch('/data');
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+  const textPromise = response.text();
+  textPromise.then(addGreetingToDom);
+}
+
+/** Adds the greeting to the DOM. */
+function addGreetingToDom(greeting) {
+  console.log('Adding greeting: '+greeting);
+  const greetingContainer = document.getElementById('greeting-container');
+  greetingContainer.innerText = greeting;
+}
```

Comment:
It's fine to merge these two methods into one I think unless you have plan to do some more things in `handleResponse` or `addGreetingToDom`. For example you could probably check for server error in the `handleResponse` method, log something if some error happens, and it would be more appropriate to have separate methods in that case.

Reply:
Thank you for the feedback. The tutorial mentioned that they can be combined into one, but as I'm still learning how it all works, the tutorial said it was okay to keep them separate if it made it clearer for me. But maybe I'll refer back to this once I'm a bit more confident with these server interactions. Thanks again!